src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -196,7 +199,60 @@ void PreserveInvoker (TypeDefinition type)
 			PreserveConstructors (type, invoker);
 			PreserveIntPtrConstructor (invoker);
 			PreserveInterfaceMethods (type, invoker);
+#if ILLINK
+			PreserveInterfaces (invoker);
+#endif // ILLINK
+		}
+
+#if ILLINK
+		void PreserveInterfaces (TypeDefinition type)
+		{
+			if (!type.HasInterfaces)
+				return;
+
+			if (!ShouldPreserveInterfaces (type))
+				return;
+
+			foreach (var iface in type.Interfaces) {
```

Comment:
Just had one more thought about this: we need to preserve *Java* interfaces, but do we need to preserve *non*-Java interfaces?

```csharp
[Register (…)]
partial interface IMyJavaInterface : IJavaPeerable {
    // …
}

partial interface IMyManagedInterface {
}

class MyClass : Java.Lang.Object, IMyManagedInterface, IMyJavaInterface {
}
```

*If* the linker determines that *nothing* uses `IMyManagedInterface`, should we be preserving it?  Or should we preserve *only* `IMyJavaInterface`?

#7097 doesn't trigger this question; it would only be relevant if *all* assemblies are linked, and customers want to get the smallest possible output.